### PR TITLE
fix: expose a way to get prompt template #14

### LINKED ISSUE #14: expose a way to get prompt template
We already have exposed a method to get prompt template by name in our python sdk. We need a replica of the same functionality in our js library.

### PR DESCRIPTION
**Pull request by abubakarsohail**
fix #14

This PR adds support for fetching a prompt from PromptLayer.

### CODE REVIEW

File: src/prompts/index.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { prompt } from "@/types";
+import { promptLayerGetPrompt, promptLayerPublishPrompt } from "@/utils";
+/**
+ * Get a prompt template from PromptLayer.
+ */
+const getPrompt = async (params: prompt.Retrieve): Promise<prompt.Response> => {
```
Comment (jzone3) [inline src/prompts/index.ts:6]: Why are the types `prompt.Retrieve` or `prompt.Response`? Why is the folder called `prompt`?

It should be called `promptTemplate` if I understand correctly?

CleanShot 2023-10-24 at 23 12 43

This tooltip isn't that descriptive to me when it's called "Response"

File: src/utils/index.ts [hunk 2]
```diff
@@ -53,6 +53,77 @@ const promptLayerApiRequest = async (body: TrackRequest) => {
...
+  } catch (e) {
+    throw new Error(
+      `PromptLayer had the following error while getting your prompt: ${e}`
+    );
+  }
+  const data = await response.json();
+  if (response.status !== 200) {
+    throwOnBadResponse(
+      data,
+      `PromptLayer had the following error while getting your prompt `
```
Comment (jzone3) [inline src/utils/index.ts:88]: should be "retrieving your prompt template" not "getting your prompt

File: src/utils/index.ts [hunk 3]
```diff
@@ -53,6 +53,77 @@ const promptLayerApiRequest = async (body: TrackRequest) => {
...
+  } catch (e) {
+    throw new Error(
+      `PromptLayer had the following error while publishing your prompt: ${e}`
+    );
+  }
+  const data = await response.json();
+  if (response.status !== 200) {
+    throwOnBadResponse(
+      data,
+      `PromptLayer had the following error while publishing your prompt`
```
Comment (jzone3) [inline src/utils/index.ts:121]: "prompt template"

**Review by jzone3 — CHANGES_REQUESTED** [review]

File: src/prompts/index.ts (hunk 1 above)
Comment (abubakarsohail) [reply src/prompts/index.ts:6]: I have updated to get types from a single types file. Will update the rest of the PRs as well.

File: src/utils/index.ts (hunk 2 above)
Comment (abubakarsohail) [reply src/utils/index.ts:88]: Updated.

File: src/utils/index.ts (hunk 3 above)
Comment (abubakarsohail) [reply src/utils/index.ts:121]: Updated.

**Review by jzone3 — APPROVED** [review]

**Pull request merged**